daffodil-runtime1/src/main/scala/org/apache/daffodil/dpath/DFDLXFunctions.scala

```diff
@@ -98,6 +98,79 @@ case class DFDLXRightShift(recipes: List[CompiledDPath], argType: NodeInfo.Kind)
+      case NodeInfo.Short => (arg1.getShort ^ arg2.getShort).toShort
+      case NodeInfo.Byte => (arg1.getByte ^ arg2.getByte).toByte
+      case NodeInfo.UnsignedLong => arg1.getBigInt.xor(arg2.getBigInt)
+      case NodeInfo.UnsignedInt => arg1.getLong ^ arg2.getLong
+      case NodeInfo.UnsignedShort => arg1.getInt ^ arg2.getInt
+      case NodeInfo.UnsignedByte => (arg1.getShort ^ arg2.getShort).toShort
+          //$COVERAGE-OFF$
+      case _ => Assert.invariantFailed(s"dfdlx:XOR not supported")
+          // $COVERAGE-ON$
+    }
+    }
+  }
+case class DFDLXAnd(recipes: List[CompiledDPath], argType: NodeInfo.Kind) extends FNTwoArgs(recipes) {
+  override def computeValue(arg1: DataValuePrimitive, arg2: DataValuePrimitive, dstate: DState): DataValuePrimitive = {
+    argType match {
+      case NodeInfo.Long => arg1.getLong & arg2.getLong
+      case NodeInfo.Int => arg1.getInt & arg2.getInt
+      case NodeInfo.Short => (arg1.getShort & arg2.getShort).toShort
+      case NodeInfo.Byte => (arg1.getByte & arg2.getByte).toByte
+      case NodeInfo.UnsignedLong => arg1.getBigInt.and(arg2.getBigInt)
+      case NodeInfo.UnsignedInt => arg1.getLong & arg2.getLong
+      case NodeInfo.UnsignedShort => arg1.getInt & arg2.getInt
+      case NodeInfo.UnsignedByte => (arg1.getShort & arg2.getShort).toShort
+      //$COVERAGE-OFF$
+      case _ => Assert.invariantFailed(s"dfdlx:AND not supported")
+      // $COVERAGE-ON$
+    }
+  }
+}
+
+case class DFDLXOr(recipes: List[CompiledDPath], argType: NodeInfo.Kind) extends FNTwoArgs(recipes) {
+  override def computeValue(arg1: DataValuePrimitive, arg2: DataValuePrimitive, dstate: DState): DataValuePrimitive = {
+    argType match {
+      case NodeInfo.Long => arg1.getLong | arg2.getLong
+      case NodeInfo.Int => arg1.getInt | arg2.getInt
+      case NodeInfo.Short => (arg1.getShort | arg2.getShort).toShort
+      case NodeInfo.Byte => (arg1.getByte | arg2.getByte).toByte
+      case NodeInfo.UnsignedLong => arg1.getBigInt.or(arg2.getBigInt)
+      case NodeInfo.UnsignedInt => arg1.getLong | arg2.getLong
+      case NodeInfo.UnsignedShort => arg1.getInt | arg2.getInt
+      case NodeInfo.UnsignedByte => (arg1.getShort | arg2.getShort).toShort
+      //$COVERAGE-OFF$
+      case _ => Assert.invariantFailed(s"dfdlx:OR not supported")
+      // $COVERAGE-ON$
+    }
+  }
+}
+
+case class DFDLXNot(recipes: CompiledDPath, argType: NodeInfo.Kind) extends FNOneArg(recipes,argType) {
+  override def computeValue(arg1: DataValuePrimitive, dstate: DState): DataValuePrimitive = {
+    argType match {
+      case NodeInfo.Long => ~arg1.getLong
+      case NodeInfo.Int => ~arg1.getInt
+      case NodeInfo.Short => ~arg1.getShort
+      case NodeInfo.Byte => ~arg1.getByte
+      case NodeInfo.UnsignedLong => arg1.getBigInt.not()
+      case NodeInfo.UnsignedInt => ~arg1.getLong
+      case NodeInfo.UnsignedShort => ~arg1.getInt
+      case NodeInfo.UnsignedByte => ~arg1.getShort
```

Comment:
I *think* we need slightly different logic for each of the Unsigned operations, simiarly to what we needed to do for shifting. For example, say we have an xs:unsignedShort with a value of `1`. That will be represented here by a java Int. And `~1` is `-2`, so we end up with a negative value for something that should be unsigned.

I think we need to get the value, convert it to the Unsigned type, perform the not operation on that, and then convert back to the right type for the infoset. So for example, for UnsignedInt I think we want something like:

```scala
case NodeInfo.UnsignedInt => (~UInt(arg1.getLong.toInt)).toLong
```
I've only done a couple simple tests, but this seems to work for what I've tested, and follows a similar pattern to what we came up with for shiftLeft/right. We'll need something similar for each of the Unsigned types.

We'll definitely want some tests to make sure this is working as expected.